### PR TITLE
[feature] #1902: Bare metal 4-peer setup script.

### LINKED ISSUE #1726: Revive `docker_test`
Currently, the `test_docker` is non-functional and has been since we upgraded to `rust 1.56`. 

As we plan to move to test on physically distributed peers, this is not a priority, but reviving this test (and figuring out how to do such tests in the future) might come in handy.

### LINKED ISSUE #1902: Create a bare-metal analogue of docker-compose
Docker compose is not always practical to use. We often need to deploy a bare-metal installation of Iroha for testing and profiling, so it would be nice to have a `scripts/deploy_4_peers.sh` to quickly check some performance metrics.

### PR DESCRIPTION
**Pull request by appetrosyan**
Signed-off-by: Aleksandr <a-p-petrosyan@yandex.ru>

### Description of the Change

Refactored `scripts` to be more self-contained and flexible for use in GitHub actions. 

Added a version of `setup_test_env.sh` that does not require `docker-compose` and uses the debug build of Iroha.

### Issue

Closes #1902.

### Benefits

Can run tests using live iroha peers without use of Docker. Creates pathway to resolve #1726 

### Possible Drawbacks

More logic relegated to bash. 

### TODO

Cleanup. 

Test on other configurations.

### CODE REVIEW

**Review by mversic — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,6 @@ cmake-build*
 \#*
 /test_docker
 /cargo-timing*.html
+/*.log
+/test/.*.json
```
Comment (mversic) [inline .gitignore:49]: superfluous?

File: scripts/setup_test_env.sh [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    export IROHA2_GENESIS_PATH
+}
+
+function run_peer () {
+    TORII_P2P_ADDR="$HOST:${p2p_ports[$1]}"
+    TORII_API_URL="$HOST:${api_ports[$1]}"
+    TORII_TELEMETRY_URL="$HOST:${telemetry_ports[$1]}"
+    IROHA_PUBLIC_KEY=${public_keys[$1]}
+    IROHA_PRIVATE_KEY="{ \"digest_function\": \"ed25519\", \"payload\": \"${private_keys[$1]}\" }"
+    exec -a "$1" "$TEST/peers/iroha" > "$1.log" "$2" & disown
```
Comment (mversic) [inline scripts/setup_test_env.sh:92]:
```suggestion
    exec -a "$1" "$TEST/peers/iroha" "$2" > "$1.log" & disown
```
can you do `exec` multiple times in a script?

Comment (appetrosyan) [reply scripts/setup_test_env.sh:92]: You can if you disown the PID.

**Review by mversic — APPROVED** [review]

**Review by s8sato — COMMENTED** [review]
Is it intentional that peers share the only storage (`./blocks`) ?

File: scripts/API/java.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+case $1 in
+    setup)
+        mkdir iroha_api_test
+        cd iroha_api_test || exit
+        git clone https://github.com/hyperledger/iroha-java.git
+        cd iroha-java || exit
+        git checkout iroha2-java
+    ;;
+    run)
+        ./gradlew build
```
Comment (s8sato) [inline scripts/API/java.sh:11]:
```suggestion
        cd iroha_api_test/iroha-java
        ./gradlew build
```

File: scripts/API/java.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+case $1 in
+    setup)
+        mkdir iroha_api_test
+        cd iroha_api_test || exit
+        git clone https://github.com/hyperledger/iroha-java.git
+        cd iroha-java || exit
+        git checkout iroha2-java
```
Comment (s8sato) [inline scripts/API/java.sh:8]: image

Does the following work?

```suggestion
        git clone https://github.com/hyperledger/iroha-java.git -b iroha2-java
        cd iroha-java || exit
```
The `iroha2-java` branch currently does not exist though

**Review by mversic — APPROVED** [review]

**Review by s8sato — APPROVED** [review]

**Pull request merged**